src/client/components/Meal.js

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { useParams } from 'react-router-dom';
+import {useState} from "react";
+import Alert from 'react-popup-alert'
+
+
+const API_URL="/api";
+
+
+function Meal({oneMeal , updateReservation, allReservations}){
+
+  let params = useParams();
+  let meal = oneMeal(parseInt(params.id));
+
+  const [alert, setAlert] =useState({
+    type: 'error',
+    text: 'This is a alert message',
+    show: false
+  })
+
+  function onCloseAlert() {
+    setAlert({
+      type: '',
+      text: '',
+      show: false
+    })
+  }
+
+  function onShowAlert(type) {
+    setAlert({
+      type: type,
+      text: 'Demo alert',
+      show: true
+    })
+  }
+
+
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    const url = `${API_URL}/reservations/`;
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({number_of_guests:parseInt(numberOfGuests),meal_id:parseInt(params.id),contact_phonenumber:phoneNumber,contact_name:name,contact_email:email}),
+    })
+      .then((reponse) => reponse.json())
+      .then(() => updateReservation(response));
+  }
+  function getReservations(mealid) {
+
+    let numberOfPeople = 0;
+   for (let i = 0; i < allReservations.length; i++) {
+   if(allReservations[i].id == mealid) {
+  numberOfPeople += allReservations[i].numberOfPeople
+}
+   }
+   return numberOfPeople < meal.max_reservations;
+ }
+ 
+  const [numberOfGuests ,setnumberOfGuests]=useState("");
+  const [phoneNumber,setPhoneNumber]=useState("");
+  const [name,setName]=useState("");
+  const [email,setEmail]=useState("");
```

Comment:
Try to keep all the useState variables in the top of your component, where you have the alert, and then have the functions.
It's a good practice to keep them all together

Reply:
doesnt work there